frame/uniques/src/lib.rs

```diff
@@ -0,0 +1,1158 @@
+				let permitted = &check == &class_details.admin || &check == &details.owner;
+				ensure!(permitted, Error::<T, I>::NoPermission);
+			}
+			let maybe_check_delegate = maybe_check_delegate.map(T::Lookup::lookup).transpose()?;
+			let old = details.approved.take().ok_or(Error::<T, I>::NoDelegate)?;
+			if let Some(check_delegate) = maybe_check_delegate {
+				ensure!(check_delegate == old, Error::<T, I>::WrongDelegate);
+			}
+
+			Asset::<T, I>::insert(&class, &instance, &details);
+			Self::deposit_event(Event::ApprovalCancelled(class, instance, details.owner, old));
+
+			Ok(())
+		}
+
+		/// Alter the attributes of a given asset.
+		///
+		/// Origin must be `ForceOrigin`.
+		///
+		/// - `class`: The identifier of the asset.
+		/// - `owner`: The new Owner of this asset.
+		/// - `issuer`: The new Issuer of this asset.
+		/// - `admin`: The new Admin of this asset.
+		/// - `freezer`: The new Freezer of this asset.
+		/// - `free_holding`: Whether a deposit is taken for holding an instance of this asset
+		///   class.
+		/// - `is_frozen`: Whether this asset class is frozen except for permissioned/admin
+		/// instructions.
+		///
+		/// Emits `AssetStatusChanged` with the identity of the asset.
+		///
+		/// Weight: `O(1)`
+		#[pallet::weight(T::WeightInfo::force_asset_status())]
+		pub(super) fn force_asset_status(
+			origin: OriginFor<T>,
+			#[pallet::compact] class: T::ClassId,
+			owner: <T::Lookup as StaticLookup>::Source,
+			issuer: <T::Lookup as StaticLookup>::Source,
+			admin: <T::Lookup as StaticLookup>::Source,
+			freezer: <T::Lookup as StaticLookup>::Source,
+			free_holding: bool,
+			is_frozen: bool,
+		) -> DispatchResult {
+			T::ForceOrigin::ensure_origin(origin)?;
+
+			Class::<T, I>::try_mutate(class, |maybe_asset| {
+				let mut asset = maybe_asset.take().ok_or(Error::<T, I>::Unknown)?;
+				asset.owner = T::Lookup::lookup(owner)?;
+				asset.issuer = T::Lookup::lookup(issuer)?;
+				asset.admin = T::Lookup::lookup(admin)?;
+				asset.freezer = T::Lookup::lookup(freezer)?;
+				asset.free_holding = free_holding;
+				asset.is_frozen = is_frozen;
+				*maybe_asset = Some(asset);
+
+				Self::deposit_event(Event::AssetStatusChanged(class));
+				Ok(())
+			})
+		}
+
+		/// Set the metadata for an asset instance.
+		///
+		/// Origin must be either `ForceOrigin` or Signed and the sender should be the Owner of the
+		/// asset `class`.
+		///
+		/// If the origin is Signed, then funds of signer are reserved according to the formula:
+		/// `MetadataDepositBase + MetadataDepositPerByte * (name.len + info.len)` taking into
+		/// account any already reserved funds.
+		///
+		/// - `class`: The identifier of the asset class whose instance's metadata to set.
+		/// - `instance`: The identifier of the asset instance whose metadata to set.
+		/// - `name`: The user visible name of this asset. Limited in length by `StringLimit`.
+		/// - `info`: The general information of this asset. Limited in length by `StringLimit`.
```

Comment:
In my opinion those two fields could be combined in one `metadata` or `data` field, because I don't think `name` is a consensus critical metadata and must live on-chain, the [ERC1155 standard](https://eips.ethereum.org/EIPS/eip-1155#rationale) also removed the `symbol` and the `name` definitions.

> The name function (for human-readable asset names, on-chain) was removed from the standard to allow the Metadata JSON to be the definitive asset name and reduce duplication of data. This also allows localization for names, which would otherwise be prohibitively expensive if each language string was stored on-chain, not to mention bloating the standard interface.

Reply:
Done.